haystack_experimental/components/embedders/sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,335 @@
+                config_kwargs=self.config_kwargs,
+                backend=self.backend,
+            )
+            if self.tokenizer_kwargs and self.tokenizer_kwargs.get("model_max_length"):
+                self._embedding_backend.model.max_seq_length = self.tokenizer_kwargs["model_max_length"]
+
+    def _validate_image_paths(self, documents: List[Document]) -> List[Dict[str, str]]:
+        """
+        Validates the image paths in the documents.
+
+        :returns:
+            A list of dictionaries, each dictionary containing the path, type, and page number of the image (for PDF).
+        :raises ValueError: If the document is missing the file_path_meta_field key in its metadata, the file path is
+            invalid, the MIME type is not supported, or the page number is missing for a PDF document.
+        """
+        images_paths_types = []
+        for doc in documents:
+            file_path = doc.meta.get(self.file_path_meta_field)
+            if file_path is None:
+                raise ValueError(
+                    f"Document with ID '{doc.id}' is missing the '{self.file_path_meta_field}' key in its metadata."
+                    f" Please ensure that the documents you are trying to convert have this key set."
+                )
+
+            resolved_file_path = Path(self.root_path, file_path)
+            if not resolved_file_path.is_file():
+                raise ValueError(
+                    f"Document with ID '{doc.id}' has an invalid file path '{resolved_file_path}'. "
+                    f"Please ensure that the documents you are trying to convert have valid file paths."
+                )
+
+            mime_type = doc.meta.get("mime_type") or mimetypes.guess_type(resolved_file_path)[0]
+            if mime_type not in IMAGE_MIME_TYPES:
+                raise ValueError(
+                    f"Document with file path '{resolved_file_path}' has an unsupported MIME type '{mime_type}'. "
+                    f"Please ensure that the documents you are trying to convert are of the supported "
+                    f"types: {', '.join(IMAGE_MIME_TYPES)}."
+                )
+
+            # If mimetype is PDF we also need the page number to be able to convert the right page
+            if mime_type == "application/pdf":
+                page_number = doc.meta.get("page_number")
+                if page_number is None:
+                    raise ValueError(
+                        f"Document with ID '{doc.id}' comes from the PDF file '{resolved_file_path}' but is missing "
+                        f"the 'page_number' key in its metadata. Please ensure that PDF documents you are trying to "
+                        f"convert have this key set."
+                    )
+                images_paths_types.append({"path": resolved_file_path, "type": "pdf", "page_number": page_number})
+            else:
+                images_paths_types.append({"path": resolved_file_path, "type": "image"})
```

Comment:
I realize it's probably a bit more clunky to work with but I wonder if we should store `mimetype` instead of a new `type`. Then at run time if the `mimetype` is in the supported `IMAGE_MIME_TYPES` then we can directly load using PIL.

Reply:
At this point, we determined if the file is an image or a PDF using mimetypes (or we raised an error).
Since we already know this information, I find it clearer to temporarily store this type information and take actions based on that. But I might miss some relevant aspects...

Reply:
Yeah fair enough. At first I thought we were also exposing this information in the component's output but I can see this is completely internal. Then totally fine to leave as is.